pymc/distributions/multivariate.py

```diff
@@ -570,14 +570,26 @@ def dist(cls, n, p, *args, **kwargs):
 
     def moment(rv, size, n, p):
         n = pt.shape_padright(n)
-        mode = pt.round(n * p)
+        mean = n * p
+        mode = pt.round(mean)
+        # Add correction term between n and approximation.
+        # We modify highest expected entry to minimize chances of negative values.
         diff = n - pt.sum(mode, axis=-1, keepdims=True)
-        inc_bool_arr = pt.abs(diff) > 0
-        mode = pt.inc_subtensor(mode[inc_bool_arr.nonzero()], diff[inc_bool_arr.nonzero()])
+        max_elem_idx = pt.argmax(mean, axis=-1, keepdims=True)
+        mode = pt.inc_subtensor(
+            pt.take_along_axis(mode, max_elem_idx, axis=-1),
+            diff,
+        )
         if not rv_size_is_none(size):
             output_size = pt.concatenate([size, [p.shape[-1]]])
             mode = pt.full(output_size, mode)
-        return mode
+        return Assert(
+            """
+        Negative value in moment found, it`s a result of
+        a known limitation of Multinomial distribution.
+        Please provide initial value manualy.
```

Comment:
This will have indentation in the error message, because of the triple quotes, no? If so, we should instead do:
```python
(
   "Error line 1. "
   "Error line 2. "
   "Error line 3."
 )
 ```

Reply:
Just the nitpick above about not using triple quotes for the error message